workers/social/lib/social/models/group/index.coffee

```diff
@@ -1545,65 +1545,117 @@ module.exports = class JGroup extends Module
-          JName.one { name: @slug }, (err, name) ->
-            removeHelper name, err, next
+          errors.push { label, err }  if err
+          next()
 
-        (next) =>
-          @fetchPermissionSet (err, permSet) ->
-            removeHelper permSet, err, next
+        removeHelper = (model, err, next, label) ->
 
-        (next) =>
-          @fetchDefaultPermissionSet (err, permSet) ->
-            removeHelper permSet, err, next
+          if err or not model
+            return kallback label, next, err
 
-        (next) =>
-          JInvitation = require '../invitation'
-          JInvitation.remove { groupName: @slug }, (err) ->
-            next err
+          model.remove (err) -> kallback label, next, err
 
-        (next) =>
-          ComputeProvider = require '../computeproviders/computeprovider'
-          ComputeProvider.destroyGroupResources this, -> next()
+        index = 0
 
-        (next) =>
-          JSession = require '../session'
-          @sendNotification 'GroupDestroyed', @slug, =>
-            JSession.remove { groupName: @slug }, (err) ->
-              next err
+        queue = [
 
-        (next) =>
-          @constructor.emit 'GroupDestroyed', this
-          next()
+          (next) ->
+            JName.one { name: slug }, (err, name) ->
+              removeHelper name, err, next, 'JName'
+
+          (next) =>
+            @fetchPermissionSet (err, permSet) ->
+              removeHelper permSet, err, next, 'PermissionSet'
+
+          (next) =>
+            @fetchDefaultPermissionSet (err, permSet) ->
+              removeHelper permSet, err, next, 'DefaultPermissionSet'
+
+          (next) ->
+            JInvitation = require '../invitation'
+            JInvitation.remove { groupName: slug }, (err) ->
+              kallback 'JInvitation', next, err
+
+          (next) =>
+            ComputeProvider = require '../computeproviders/computeprovider'
+            ComputeProvider.destroyGroupResources this, -> next()
+
+          (next) ->
+            url = '/api/social/payment/subscription/delete'
+            { deleteReq } = require '../socialapi/requests'
+
+            deleteReq url, { sessionToken }, (err, body) ->
+              err = null  if err?.description is 'not found'
+              kallback 'Subscription', next, err
+
+          (next) ->
+            JApiToken = require '../apitoken'
+            JApiToken.remove { group: slug }, (err) ->
+              kallback 'JApiToken', next, err
+
+          (next) =>
+            JSession = require '../session'
+            @sendNotification 'GroupDestroyed'
+            JSession.remove { groupName: slug }, (err) ->
+              kallback 'JSession', next, err
+
+          (next) =>
+            @remove (err) -> kallback 'JGroup', next, err
+
+        ]
+
+        do deleteTeam = =>
+
+          async.series queue, =>
+
+            callback()  if index is 0
+            return  unless errors.length
+
+            index++
```

Comment:
this retry wont help us on the intermittent errors, please use retry with backoff capabilities. please see client/app/lib/realtimecontroller.coffee and 
client/app/lib/searchcontroller.coffee as an example